### PR TITLE
fix: FORMS-1025 ab#23188 fix chefs event service

### PR DESCRIPTION
**Pull request by jimmyPasta**
# Description
createSubmission is called in 3 scenarios
- on saving of draft submissions - click the save icon on a brand new submission
- on submit of a brand new draft submission
- on submit of a submission that is not a draft

It does not get called when updating a draft submission

This problem solves events not being fired on the update of submissions when submit is clicked.
This problem simplifies the code for event subscription into a eventService

Also fixes:
**Observations:**

The bug is that if you create two forms in a row in Chefs and enable the "event subscription" feature for both forms, then in the second form the event subscription will be  already present in the event subscription text fields but you're unable to save.

## Types of changes
Bug fix (non-breaking change which fixes an issue) 

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have run the npm script lint on the frontend and backend
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have approval from the product owner for the contribution in this pull request

## Further comments
Will walk through the code changes 
All tests here have now passed
image

### CODE REVIEW

**Comment by Stephan-McColm** [comment]
Also tested via:
image

File: app/src/forms/form/service.js [hunk 1]
```diff
@@ -685,15 +667,7 @@ const service = {
...
-      const { subscribe } = await service.readForm(formId);
-      if (subscribe && subscribe.enabled) {
-        const subscribeConfig = await service.readFormSubscriptionDetails(formId);
-        const config = Object.assign({}, subscribe, subscribeConfig);
-        const formVersion = new FormVersion();
-        formVersion.id = version.id;
-        formVersion.formId = formId;
-        service.postSubscriptionEvent(config, formVersion, null, SubscriptionEvent.FORM_DRAFT_PUBLISHED);
-      }
+      eventService.publishFormEvent(formId, version.id, version.published);
```
Comment (usingtechnology) [inline app/src/forms/form/service.js:670]: I don't see how this is sending the `FORM_DRAFT_PUBLISHED` event? `eventService.publishFormEvent` only sends the `FORM_PUBLISHED` and `FORM_UNPUBLISHED` events.

**Review by usingtechnology — CHANGES_REQUESTED** [review]
Question about DRAFT publish events... may have missed something but appears that is no longer sent.

**Review by usingtechnology — APPROVED** [review]
we can deal with a tiny clean up of unused events in another PR.

**Review by vijaivir — APPROVED** [review]

**Review by WalterMoar — APPROVED** [review]

**Pull request merged**